src/service/MosaicService.ts

```diff
@@ -62,16 +62,30 @@ export class MosaicService {
      * @returns {Observable<MosaicAmountView[]>}
      */
     mosaicsAmountView(mosaics: Mosaic[]): Observable<MosaicAmountView[]> {
-        return observableOf(mosaics).pipe(
-            mergeMap((_) => _),
-            mergeMap((mosaic: Mosaic) => this.mosaicsView([new MosaicId(mosaic.id.id.toDTO())]).pipe(
```

Comment:
this in fact will call the networkType observable for any mosaics. What about fixing the issue in this method `mosaicsView` (fix: get network type only once for whole Http service with said endpoint.) https://github.com/nemtech/nem2-sdk-typescript-javascript/blob/1e4814a1d47b6127ac5c5489c9f9ce985cb50c37/src/service/MosaicService.ts#L51

Seems like the fix is very cumbersome and could be refactored using mergeMap or other operators.

Reply:
The mosaic view looks fine to me, if you pass an array to it, it will just
call the network type once:
https://github.com/nemtech/nem2-sdk-typescript-javascript/blob/1e4814a1d47b6127ac5c5489c9f9ce985cb50c37/src/service/MosaicService.ts#L76

Yes, my proposition is not elegant... and I actually just noticed that it does not work, because getMosaics() does not return the mosaics in the same order.